packages/uif/src/Retries.ts

```diff
@@ -0,0 +1,46 @@
+import assert from 'node:assert'
+
+export interface RetryStrategy {
+  shouldRetry: () => boolean
+  timeoutMs: () => number
+  clear: () => void
+}
+
+interface ExponentialBackOffOpts {
+  stepMs: number
+  maxAttempts: number
+  maxDistanceMs?: number
+}
+
+/**
+ *
+ * @param opts.maxAttempts - use Infinity for indefinite retries
+ * @param opts.stepMs - step multiplied by 2**attempts gives the distance between retries
+ * @param opts.maxDistanceMs - maximum distance between retries (default: Infinity)
+ * @returns
+ */
+function exponentialBackOff(opts: ExponentialBackOffOpts): RetryStrategy {
+  let attempts = 0
+  const maxAttempts = opts.maxAttempts
+  assert(maxAttempts > 0)
+  const maxDistanceMs = opts.maxDistanceMs ?? Infinity
+  assert(maxDistanceMs > 0)
+
+  return {
+    shouldRetry: () => {
+      attempts++
```

Comment:
Maybe this function should be indempotent? And we can have another function `markAttempt`?

Reply:
5f2b23a